docs/using-the-compiler.rst

```diff
@@ -0,0 +1,269 @@
+          // Mandatory: Error type, such as "TypeError", "InternalCompilerError", "Exception", etc
+          type: "TypeError",
+          // Mandatory: Component where the error originated.
+          component: "general", "why3", "ewasm", etc.
+          // Mandatory
+          severity: "warning" or "error",
+          // Mandatory
+          message: "Invalid keyword"
+        }
+      ],
+      // This contains the file-level outputs. In can be limited/filtered by the outputSelection settings.
+      sources: {
+        "sourceFile.sol": {
+          // Identifier (used in source maps)
+          id: 1,
+          // The AST object
+          ast: {}
+        }
+      },
+      // This contains the contract-level outputs. It can be limited/filtered by the outputSelection settings.
+      contracts: {
+        "sourceFile.sol": {
+          "ContractName": {
+            // The Ethereum Contract ABI. If empty, it is represented as an empty array.
+            // See https://github.com/ethereum/wiki/wiki/Ethereum-Contract-ABI
+            abi: [],
+            evm: {
+              // Intermediate representation (string)
+              ir: "",
+              // Assembly (string)
+              assembly: "",
+              // Old-style assembly (string)
+              asm: "",
+              // Old-style assembly (JSON object)
+              asmJSON: [],
+              bytecode: {
+                // The bytecode as a hex string.
+                object: "00fe",
+                // The source mapping as a string. See the source mapping definition.
+                sourceMap: "",
+                // If given, this is an unlinked object (cannot be filtered out explicitly, might be
+                // filtered if both bytecode, runtimeBytecode, opcodes and others are filtered out)
+                linkReferences: {
+                  "libraryFile.sol": {
+                    // Byte offsets into the bytecode. Linking replaces the 20 bytes located there.
+                    "Library1": [
+                      { start: 0, end: 20 },
+                      { start: 200, end: 220 }
+                    ]
+                  }
+                }
+              }
+              // The same layout as above.
+              deployedBytecode: { },
+              // Opcodes list (string)
+              opcodes: "",
+              // The list of function hashes
+              methodIdentifiers: {
+                "5c19a95c": "delegate(address)",
+              },
+              // Function gas estimates
+              gasEstimates: {
+                creation: {
+                  dataCost: 420000,
+                  // -1 means infinite (aka. unknown)
+                  executionCost: -1
+                external: {
+                  "delegate(address)": 25000
+                },
+                internal: {
+                  "heavyLifting()": -1
+                }
+              }
+            },
+            // See the Metadata Output documentation
```

Comment:
Cannot easily put links within the documentation since that part is a code piece. We need to think about how to nicely document this.